### PR TITLE
Create JSON data bundle regenerator #6045

### LINKED ISSUE #6045: Create JSON data bundle regenerator
Notice that it's **re**generator instead of generator, meaning it will just read existing data and regenerate it. Advantages:
- Standardized coding standard
- Duplicate fields will be caught
- Outdated fields will be rid of
- Missing fields will be updated

This has been field-tested and the number of violations is unbelievable.

### PR DESCRIPTION
**Pull request by wkurniawan07**
Fixes #6045 

- The script has been run 10 times in succession and no changes occur to any of the JSON files.
- Some implementations have to change (in particular `Hash*` -> `LinkedHash*`, `transient` for non-serialized attributes) in order to maintain insertion orders etc. Add: also, some attributes are re-ordered to minimize the lines changed.
- Worry not about the files that cannot be displayed (even after `?w=1`), because they are simply regenerations and the workings of it can be seen in countless other files.

### CODE REVIEW

**Comment by wkurniawan07** [comment]
@JosephineKwa I suppose you're busy at the moment? I can assign somebody else if that's the case.

**Comment by JosephineKwa** [comment]
Yup would be good if you could assign somebody else :pensive:, thanks!

**Comment by junhaoyap** [comment]
Will take a look tomorrow after looking at the issue to know what it's supposed to do, it's too late for that now at 4am :p

**Comment by wkurniawan07** [comment]
@junhaoyap reminder on this one :p

**Comment by junhaoyap** [comment]
My bad - hackathon and nightless sleep (I think I mean sleepless nights) - on it now

**Comment by junhaoyap** [comment]
Ignoring the commit that has 17k changes (2nd commit), looking at the rest :)

**Comment by junhaoyap** [comment]
For the second point on changing to ```Map... new LinkedHashMap```, I've seen in many places that it's a common convention to have it that way (makes it much easier to change implementations in the future in situations such as this), unsure why we had it as ```HashMap... new HashMap``` in the first place, is it something in our guidelines?

**Comment by junhaoyap** [comment]
Like the usage of unknown types <?> too 👍

**Comment by junhaoyap** [comment]
1. Question - Why are we ```disabling htmlEscaping```?
2. For ```&& !file.getName().startsWith("feedbackSessionTeamEvaluationTemplate")``` can we instead use a list of strings to check against that we should ignore to make it more future proof? 
3. Question - We are adding ```transient``` so that they don't get regenerated into the JSON, just to check my understanding, correct me if I'm wrong

Otherwise, looks good

**Comment by junhaoyap** [comment]
I'll set it back to ongoing, just some questions and a minor comment, let me know once I should look at it again, hopefully I won't miss it (no more hackathons on this weekend ;))

**Comment by wkurniawan07** [comment]
`Map<String, String> a = new HashMap<String, String>()`, `List<String> b = new ArrayList<String>()` is recommended even by PMD for the exact same reason (look up the Loose Coupling rule). It was not anywhere in our Java style guide (and if it is, it's about time to revamp it).

1. Otherwise we cannot store `'`, `"`, `<`, `>`. They will otherwise be stored as their unicode counterparts (e.g `'` will be `\u0027`).
2. Possible, will be done.
3. That's right.

**Comment by junhaoyap** [comment]
@wkurniawan07 it's apparently not in it, let's cc @damithc here regarding that so we can have a discussion

**Comment by junhaoyap** [comment]
1. Got it
2. Looked at the changes, ok
3. Thank you for the clarification

At this point it's good to go and I'll still like to hold the discussion regarding the coupling in case of future needed changes in other parts of the project

File: src/client/java/teammates/client/scripts/DataBundleRegenerator.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    );
+    
+    private DataBundleRegenerator() {
+        // script-like, not meant to be instantiated
+    }
+    
+    private static void regenerateDataBundleJson(File folder) throws IOException {
+        File[] listOfFiles = folder.listFiles();
+        for (File file : listOfFiles) {
+            if (file.getName().endsWith(".json") && !NON_DATA_BUNDLE_JSON.contains(file.getName())) {
```
Comment (damithc) [inline src/client/java/teammates/client/scripts/DataBundleRegenerator.java]: Reverse condition to reduce nesting. 
```java
if(!condition) { continue;}
```

**Comment by damithc** [comment]
Cosmetic change requested. Can merge after that.

**Pull request merged**